test/system/src/test/java/io/pravega/test/system/KeyValueStoreTest.java

```diff
@@ -0,0 +1,744 @@
+package io.pravega.test.system;
+
+import io.pravega.client.ClientConfig;
+import io.pravega.client.KeyValueTableFactory;
+import io.pravega.client.admin.KeyValueTableInfo;
+import io.pravega.client.admin.KeyValueTableManager;
+import io.pravega.client.control.impl.Controller;
+import io.pravega.client.control.impl.ControllerImpl;
+import io.pravega.client.control.impl.ControllerImplConfig;
+import io.pravega.client.netty.impl.ConnectionFactory;
+import io.pravega.client.netty.impl.ConnectionFactoryImpl;
+import io.pravega.client.stream.Serializer;
+import io.pravega.client.stream.impl.UTF8StringSerializer;
+import io.pravega.client.tables.*;
+import io.pravega.client.tables.impl.KeyValueTableFactoryImpl;
+import io.pravega.test.common.AssertExtensions;
+import io.pravega.test.system.framework.SystemTestRunner;
+import io.pravega.test.system.framework.Utils;
+import io.pravega.test.system.framework.services.Service;
+import lombok.Cleanup;
+import lombok.extern.slf4j.Slf4j;
+import org.junit.*;
+import org.junit.runner.RunWith;
+
+import java.net.URI;
+import java.nio.ByteBuffer;
+import java.util.*;
+import java.util.concurrent.*;
+import java.util.stream.Collectors;
+
+import static org.apache.commons.lang.RandomStringUtils.randomAlphanumeric;
+import static org.junit.Assert.*;
+
+@Slf4j
+@RunWith(SystemTestRunner.class)
+public class KeyValueStoreTest extends AbstractSystemTest {
+
+    private static final String SCOPE_NAME = "SampleScope3" + randomAlphanumeric(5);
+    private static final String KVT_NAME = "TestKVT1";
+    private static final KeyValueTableInfo KVT = new KeyValueTableInfo(SCOPE_NAME,KVT_NAME);
+    private static final KeyValueTableConfiguration config = KeyValueTableConfiguration.builder().partitionCount(2).build();
+    private static final Serializer<Integer> KEY_SERIALIZER = new IntegerSerializer();
+    private static final Serializer<String> VALUE_SERIALIZER = new UTF8StringSerializer();
+    private URI controllerURI = null;
+    private Controller controller = null;
+    private KeyValueTableFactory keyValueTableFactory;
+    private KeyValueTable<Integer, String> keyValueTable;
+    private ConnectionFactory connectionFactory;
+    private Service controllerInstance;
+
+    @Before
+    public void setup() throws Exception {
+        controllerInstance = Utils.createPravegaControllerService(null);
+        List<URI> ctlURIs = controllerInstance.getServiceDetails();
+        final List<String> uris = ctlURIs.stream().filter(ISGRPC).map(URI::getAuthority).collect(Collectors.toList());
+        controllerURI = URI.create("tcp://" + String.join(",", uris));
+        final ClientConfig clientConfig = Utils.buildClientConfig(controllerURI);
+        final ScheduledExecutorService controllerExecutor= Executors.newScheduledThreadPool(5);
```

Comment:
1. use ExecutorServiceHelpers.newScheduledThreadPool.
2. store this in a class variable and shut it down in `tearDown` (`ExecutorServiceHelpers.shutdown`)

Reply:
Sure, will change as said.